src/main/java/com/paskar/email/application/repositiory/EmailStorage.java

```diff
@@ -0,0 +1,49 @@
+package com.paskar.email.application.repositiory;
+
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.paskar.email.application.console.Email;
+import org.springframework.stereotype.Component;
+
+import java.io.*;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class EmailStorage implements EmailRepository {
+    private final static String baseFile = "emailList.json";
+
+    @Override
+    public void save(List<Email> email) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        PrintWriter out = new PrintWriter(new BufferedWriter(
```

Comment:
when working with IO we should release resources after we processing completed. best way to do it is to use try-with-resources:
```
try (FileWriter fw = new FileWriter(baseFile, true);
      BufferedWriter bw = new BufferedWriter(fw);
      PrintWriter pw = new PrintWriter(bw)) {
     //work with writer
}
```